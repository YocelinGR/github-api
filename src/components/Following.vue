<template>
  <div>
    <div v-for="userFollowing in user.following.nodes" :key="userFollowing.id" class="followers row">
    <div class="col-md-2">
      <img :src=" userFollowing.avatarUrl" alt="user image">
    </div>
    <div class="col md-8">
       <span class="name">{{ userFollowing.name }}</span>
       <span class="userName">{{ userFollowing.login }}</span>
       <p class="userName">{{ userFollowing.bio}}</p>
       <p v-if="userFollowing.location" class="userName"><i class='fas fa-map-marker-alt'></i>
        {{  userFollowing.location }}</p>
    </div>
    <div class="col-md-2">
      <button class="btn btn-light">Unfollow</button>
    </div>
  </div>
  </div>
</template>

<script>
import gql from 'graphql-tag';

const following = gql`
  query user {
  user(login: "YocelinGR"){
    id
    following(first: 20) {
      nodes {
        id
        login
        name
        avatarUrl(size: 64)
        bio
        location
      }
    }
  }
}
`;
export default {
  data : () => ({
    user: {},
  }),
  apollo: {
    user: {
      query: following,
    }
  }
}
</script>

<style scoped>
  img {
    border-radius: 4px;
  }
  .name {
    font-size: 16px;
    color: rgb(41, 40, 40);
    padding: 6px;
  }
  p.userName{
    font-size: 14px;
    color: rgb(110, 102, 102);
    padding: 2px;
    padding-bottom: 1px !important;
  }

  .followers {
    justify-content: center;
    align-items: center;
    margin: 15px;
    padding-bottom: 10px;
    border-bottom: rgb(187, 185, 185) solid 1px;
  }
  .btn {
    background-color: #f0f1f1;
    border-color: #a5a6a7;
  }
  .btn:hover {
    background-color: #ccd2db;
  }
</style>
