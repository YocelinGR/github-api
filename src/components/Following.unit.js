import Following from './Following.vue'
import { createLocalVue } from '@vue/test-utils'

describe('@components/Following', () => {
  const localVue = createLocalVue()
  it('Following is a vue instance', () => {
    const wrapper = shallowMount(Following, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('Displays repositories that user follows', () => {
    const wrapper = shallowMount(Following, { localVue })
    wrapper.setData({
      user: {
        id: "MDQ6VXNlcjM5ODMzMDQ1",
        following: {
          nodes: [
            {
              id: "MDQ6VXNlcjE1MDMzMA==",
              login: "getify",
              name: "Kyle Simpson",
              avatarUrl: "https://avatars0.githubusercontent.com/u/150330?s=64&v=4",
              bio: "I teach JavaScript and I'd love to come help your team's developers. If that's interesting to you, please reach out to me getify@gmail.com.",
              location: "Austin, TX"
            }
          ]
        }
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
