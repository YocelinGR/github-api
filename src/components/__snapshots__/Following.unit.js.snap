// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Following Displays repositories that user follows 1`] = `
<div>
  <div
    class="followers row"
  >
    <div
      class="col-md-2"
    >
      <img
        alt="user image"
        src="https://avatars0.githubusercontent.com/u/150330?s=64&v=4"
      />
    </div>
     
    <div
      class="col md-8"
    >
      <span
        class="name"
      >
        Kyle Simpson
      </span>
       
      <span
        class="userName"
      >
        getify
      </span>
       
      <p
        class="userName"
      >
        I teach JavaScript and I'd love to come help your team's developers. If that's interesting to you, please reach out to me getify@gmail.com.
      </p>
       
      <p
        class="userName"
      >
        <i
          class="fas fa-map-marker-alt"
        />
        
      Austin, TX
      </p>
    </div>
     
    <div
      class="col-md-2"
    >
      <button
        class="btn btn-light"
      >
        Unfollow
      </button>
    </div>
  </div>
</div>
`;
